src/sentry_webhook/post_sentry_event_to_salesforce.ts

```diff
@@ -0,0 +1,162 @@
+    const formFields = isProd ? SALESFORCE_FORM_FIELDS_PROD : SALESFORCE_FORM_FIELDS_DEV;
+    const formValues = isProd ? SALESFORCE_FORM_VALUES_PROD : SALESFORCE_FORM_VALUES_DEV;
+    const isClient = project.indexOf('client') !== -1;
+    const formData =
+        getSalesforceFormData(formFields, formValues, event, event.user.email, isClient);
+    const req = https.request(
+        {
+          host: salesforceHost,
+          path: SALESFORCE_PATH,
+          protocol: 'https:',
+          method: 'post',
+          headers: {
+            // The production server will reject requests that do not specify this content type.
+            'Content-Type': 'application/x-www-form-urlencoded'
+          }
+        },
+        (res) => {
+          if (res.statusCode === 200) {
+            resolve();
+          } else {
+            reject(new Error(`Failed to post form data, response status: ${res.statusCode}`));
+          }
+        });
+    req.on('error', (err) => {
+      reject(new Error(
+          `Failed to submit form: ${err}`,
+          ));
+    });
+    req.write(formData);
+    req.end();
+  });
+}
+
+// Returns a URL-encoded string with the Salesforce form data.
+function getSalesforceFormData(
+    formFields: SalesforceFormFields, formValues: SalesforceFormValues, event: sentry.SentryEvent,
+    email: string, isClient: boolean): string {
+  const form = [];
+  form.push(encodeFormData(formFields.orgId, formValues.orgId));
+  form.push(encodeFormData(formFields.recordType, formValues.recordType));
+  form.push(encodeFormData(formFields.email, email));
+  form.push(encodeFormData(formFields.sentryEventId, event.event_id));
+  form.push(encodeFormData(formFields.description, event.message));
+  form.push(encodeFormData(formFields.type, isClient ? 'Outline client' : 'Outline manager'));
+  const tags = getTagsMap(event.tags);
+  if (!!tags) {
+    form.push(encodeFormData(formFields.category, tags.get('category')));
+    form.push(encodeFormData(formFields.os, tags.get('os.name')));
+    form.push(encodeFormData(formFields.version, tags.get('sentry:release')));
+    if (!isClient) {
+      form.push(encodeFormData(formFields.cloudProvider, tags.get('cloudProvider')));
+    }
+  }
+  return form.join('&');
+}
+
+function encodeFormData(field: string, value?: string) {
+  return `${encodeURIComponent(field)}=${encodeURIComponent(value || '')}`;
+}
+
+// Although SentryEvent.tags is declared as an index signature object, it is actually an array of
+// arrays i.e. [['key0': 'value0'], ['key1': 'value1']]. Converts the tags to a Map of strings.
+function getTagsMap(tags?: {[key: string]: string}) {
+  if (!tags) {
+    return null;
+  }
+  const tagsMap = new Map<string, string>();
+  for (const i of Object.keys(tags)) {
+    try {
+      const tagEntry = tags[i];
+      tagsMap.set(tagEntry[0], tagEntry[1]);
+    } catch (e) {
+      console.error('Failed to process tag entry');
```

Comment:
Don't swallow the exception.

Reply:
I don't want to fail the whole operation for one malformed tag. I added this as a precaution, though I have not seen malformed tags in any reports.

Reply:
Mmm, okay. Let's make it a `warn` then, and include some debugging in the message, e.g.:
```ts
console.warn(`failed to process tag entry ${tagEntry[0]}`);
```
(does the environment have `console.warn`? I forget)